### PR TITLE
:tada: Make job docker container name meaningful

### LINKED ISSUE #9584: Add container name for source/destination container launched by Airbyte worker
## Tell us about the problem you're trying to solve
Today the worker launch containers using the command:
```
2022-01-18 22:49:20 INFO i.a.w.p.DockerProcessFactory(create):171 - Preparing command: docker run --rm --init -i -v airbyte_workspace:/data -v /tmp/airbyte_local:/local -w /data/27940/0 --network host --log-driver none airbyte/destination-snowflake:0.3.24 write --config destination_config.json --catalog destination_catalog.json
``` 

we can add the tag `--name <docker-container-name>_<4-lenght-hash>`
With the name would be very easy to identify the container I need to inspect.

One example where this can be useful:
image

## Describe the solution you’d like
A clear and concise description of what you want to see happen, or the change you would like to see

## Describe the alternative you’ve considered or used
A clear and concise description of any alternative solutions or features you've considered or are using today.

## Are you willing to submit a PR?

Remove this with your answer :-)

### PR DESCRIPTION
**Pull request by davinchia**
## What
Closes https://github.com/airbytehq/airbyte/issues/9584.

This now follows the Kubernetes name convention of `<image-name>-<job-id>-<attempt-number>`.

This should make things easier for oss users doing oncall for docker deployments.

## How
Mirror the Kubernetes name convention: https://github.com/airbytehq/airbyte/blob/master/airbyte-workers/src/main/java/io/airbyte/workers/process/KubeProcessFactory.java#L180

We can probably reuse the naming functions across both processes. We are going to do more changes in this area around making things more ergonomic (e.g. instead of always using `sync`, we want to replace this with the actual operation - discover, check connection etc.) so we'll do refactoring in that PR. I want to unblock this in the time being since open source users are having difficulty debugging now.

## Recommended reading order
The one file.

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Comment by davinchia** [comment]
Evidence this works:

Screenshot 2022-05-01 at 11 58 38 PM

**Review by jdpgrailsdev — APPROVED** [review]
:shipit: with one nit.

File: airbyte-workers/src/main/java/io/airbyte/workers/process/DockerProcessFactory.java [hunk 1]
```diff
@@ -163,6 +167,28 @@ public Process create(final String jobId,
...
 
+  private static String createContainerName(final String fullImagePath, final String jobId, final int attempt) {
+    final var versionDelimiter = ":";
+    final var noVersion = fullImagePath.split(versionDelimiter)[0];
+
+    final var dockerDelimiter = "/";
+    final var nameParts = noVersion.split(dockerDelimiter);
+    var imageName = nameParts[nameParts.length - 1];
+
+    final var randSuffix = RandomStringUtils.randomAlphabetic(5).toLowerCase();
```
Comment (pmossman) [inline airbyte-workers/src/main/java/io/airbyte/workers/process/DockerProcessFactory.java:178]: Just curious, is the random suffix only necessary due to the fact that we call everything `sync`? So without the random suffix, we'd potentially have name conflicts between check/discover/etc?

**Review by pmossman — APPROVED** [review]
LGTM, +1 to Jonathan's nit

File: airbyte-workers/src/main/java/io/airbyte/workers/process/DockerProcessFactory.java [hunk 2]
```diff
@@ -163,6 +167,28 @@ public Process create(final String jobId,
...
+
+    var podName = imageName + "-" + suffix;
+    final var podNameLenLimit = 128;
+    if (podName.length() > podNameLenLimit) {
+      final var extra = podName.length() - podNameLenLimit;
+      imageName = imageName.substring(extra);
+      podName = imageName + "-" + suffix;
+    }
+
+    return podName;
```
Comment (marcosmarxm) [inline airbyte-workers/src/main/java/io/airbyte/workers/process/DockerProcessFactory.java:189]: nit but should this be called `containerName` or should keep the same for future refactor?

**Review by marcosmarxm — APPROVED** [review]

File: airbyte-workers/src/main/java/io/airbyte/workers/process/DockerProcessFactory.java (hunk 2 above)
Comment (davinchia) [reply airbyte-workers/src/main/java/io/airbyte/workers/process/DockerProcessFactory.java:189]: I'll keep it for a future refactor.

Comment (davinchia) [reply airbyte-workers/src/main/java/io/airbyte/workers/process/DockerProcessFactory.java:189]: which I'll do in the next few days.

File: airbyte-workers/src/main/java/io/airbyte/workers/process/DockerProcessFactory.java (hunk 1 above)
Comment (davinchia) [reply airbyte-workers/src/main/java/io/airbyte/workers/process/DockerProcessFactory.java:178]: Not really. The random suffix is there to make it truly random without any assumptions on the job-id/attempt-id combination e.g. that we have globally unique job-id/attempt-id combinations.

The synchronous jobs such as check, discover etc are assigned a job id that is a UUID so we can assume the names will always be random.

Comment (davinchia) [reply airbyte-workers/src/main/java/io/airbyte/workers/process/DockerProcessFactory.java:178]: Does that make sense? Happy to chat more on Slack.

File: airbyte-workers/src/main/java/io/airbyte/workers/process/DockerProcessFactory.java [hunk 3]
```diff
@@ -163,6 +167,28 @@ public Process create(final String jobId,
     }
   }
 
+  private static String createContainerName(final String fullImagePath, final String jobId, final int attempt) {
+    final var versionDelimiter = ":";
```
[the remark this answers is not in the document]
Comment (davinchia) [reply airbyte-workers/src/main/java/io/airbyte/workers/process/DockerProcessFactory.java]: Good catch. Done.

**Comment by marcosmarxm** [comment]
@Amruta-Ranade this is a good feature for OSS in the annoucement.

**Pull request merged**